bus-mapping/src/circuit_input_builder.rs

```diff
@@ -550,11 +554,25 @@ impl<'a> CircuitInputStateRef<'a> {
     /// [`RWCounter`] and then adds a reference to the stored operation
     /// ([`OperationRef`]) inside the bus-mapping instance of the current
     /// [`ExecStep`].  Then increase the block_ctx [`RWCounter`] by one.
-    pub fn push_op<T: Op>(&mut self, op: T) {
-        let op_ref = self
-            .block
-            .container
-            .insert(Operation::new(self.block_ctx.rwc.inc_pre(), op));
+    pub fn push_op<T: Op>(&mut self, rw: RW, op: T) {
+        let op_ref =
+            self.block
+                .container
+                .insert(Operation::new(self.block_ctx.rwc.inc_pre(), rw, op));
+        self.step.bus_mapping_instance.push(op_ref);
+    }
+
+    /// Push an [`Operation`] with reversible to be true into the
+    /// [`OperationContainer`] with the next [`RWCounter`] and then adds a
+    /// reference to the stored operation ([`OperationRef`]) inside the
+    /// bus-mapping instance of the current [`ExecStep`]. Then increase the
+    /// block_ctx [`RWCounter`] by one.
+    pub fn push_op_reversible<T: Op>(&mut self, rw: RW, op: T) {
```

Comment:
I can't find where this method is being called.  Maybe it should be called in `apply_reverted_op`?

Reply:
Reply from Han:
>  `push_op_reversible` will be called when the op is reversible, since no all state write needs reversion, for example, nonce increment of `BeginTx`. You can find their usage in PR of `CALL` and `EndTx`.